packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts

```diff
@@ -0,0 +1,238 @@
+
+  /**
+   * Custom configuration file, s3 or file
+   * @default - determined by checking configFileValue with S3 ARN.
+   */
+  readonly configFileType?: FirelensConfigFileType;
+
+  /**
+   * Custom configuration file, S3 ARN or a file path
+   */
+  readonly configFileValue: string;
+}
+
+/**
+ * Firelens Configuration
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-taskdef
+ */
+export interface FirelensConfig {
+
+  /**
+   * The log router to use
+   * @default - fluentbit
+   */
+  readonly type: FirelensLogRouterType;
+
+  /**
+   * Firelens options
+   * @default - no additional options
+   */
+  readonly options?: FirelensOptions;
+}
+
+/**
+ * The properties in a firelens log router.
+ */
+export interface FirelensLogRouterProps extends ContainerDefinitionProps {
+  /**
+   * Firelens configuration
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * The options for creating a firelens log router.
+ */
+export interface FirelensLogRouterDefinitionOptions extends ContainerDefinitionOptions {
+  /**
+   * Firelens configuration
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * Render to CfnTaskDefinition.FirelensConfigurationProperty from FirelensConfig
+ */
+function renderFirelensConfig(firelensConfig: FirelensConfig): CfnTaskDefinition.FirelensConfigurationProperty {
+  if (!firelensConfig.options) {
+    return { type: firelensConfig.type };
+  } else {
+    // firelensConfig.options.configFileType has been filled with s3 or file type in constructor.
+    return {
+      type: firelensConfig.type,
+      options: {
+        'enable-ecs-log-metadata': firelensConfig.options.enableECSLogMetadata ? 'true' : 'false',
+        'config-file-type': firelensConfig.options.configFileType!,
+        'config-file-value': firelensConfig.options.configFileValue
+      }
+    };
+  }
+
+}
+
+/**
+ * SSM parameters for latest fluent bit docker image in ECR
+ * https://github.com/aws/aws-for-fluent-bit#using-ssm-to-find-available-versions
+ */
+const fluentBitImageSSMPath = '/aws/service/aws-for-fluent-bit';
+
+/**
+ * Obtain Fluent Bit image in Amazon ECR and setup corresponding IAM permissions.
+ * ECR image pull permissions will be granted in task execution role.
+ * Cloudwatch logs or Firehose permissions will be grant by check options in logDriverConfig.
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-using-fluentbit
+ */
+export function obtainDefaultFluentBitECRImage(task: TaskDefinition, logDriverConfig?: LogDriverConfig, imageTag?: string): ContainerImage {
+  // grant ECR image pull permissions to executor role
+  task.addToExecutionRolePolicy(new iam.PolicyStatement({
+    actions: [
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage"
+    ],
+    resources: ['*']
+  }));
+
+  // grant cloudwatch or firehose permissions to task role
```

Comment:
add kinesis streams as well: https://github.com/aws/amazon-kinesis-streams-for-fluent-bit#permissions

Reply:
done